Bottom: 90732a346985144bed557ac18ddd1e7d31460241
Top:    3c2db26f493ab230d8eb7f9e8565c9c9b6c9214c
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-28 21:49:11 +0100

Refresh of move-preloadelementpcb-and

---

diff --git a/src/parse_y.y b/src/parse_y.y
index e6ce092..980ffab 100644
--- a/src/parse_y.y
+++ b/src/parse_y.y
@@ -215,30 +215,37 @@ parsepcb
 			}
 			   
 		| {
-		    yyFont = &yyPCB->Font;
-		    yyData = yyPCB->Data;
-		    yyData->pcb = yyPCB;
-		    yyData->LayerN = 0;
-		    layer_group_string = NULL;
+		    if (yyPCB != NULL)
+		      {
+			yyFont = &yyPCB->Font;
+			yyData = yyPCB->Data;
+			yyData->pcb = yyPCB;
+			yyData->LayerN = 0;
+			layer_group_string = NULL;
+		      }
 		  }
 		  element
 		  {
 		    PCBType *pcb_save = PCB;
 		    ElementType *e;
 
-		    LayerFlag[0] = true;
-		    LayerFlag[1] = true;
-		    yyData->LayerN = 2;
-
-		    CreateNewPCBPost (yyPCB, 0);
-		    ParseGroupString("1,c:2,s", &yyPCB->LayerGroups, yyData->LayerN);
-		    e = yyPCB->Data->Element->data; /* we know there's only one */
-		    PCB = yyPCB;
-		    MoveElementLowLevel (yyPCB->Data, e, -e->BoundingBox.X1, -e->BoundingBox.Y1);
-		    PCB = pcb_save;
-		    yyPCB->MaxWidth = e->BoundingBox.X2;
-		    yyPCB->MaxHeight = e->BoundingBox.Y2;
-		    yyPCB->is_footprint = 1;
+		    if (yyPCB != NULL)
+		      {
+
+			LayerFlag[0] = true;
+			LayerFlag[1] = true;
+			yyData->LayerN = 2;
+
+			CreateNewPCBPost (yyPCB, 0);
+			ParseGroupString("1,c:2,s", &yyPCB->LayerGroups, yyData->LayerN);
+			e = yyPCB->Data->Element->data; /* we know there's only one */
+			PCB = yyPCB;
+			MoveElementLowLevel (yyPCB->Data, e, -e->BoundingBox.X1, -e->BoundingBox.Y1);
+			PCB = pcb_save;
+			yyPCB->MaxWidth = e->BoundingBox.X2;
+			yyPCB->MaxHeight = e->BoundingBox.Y2;
+			yyPCB->is_footprint = 1;
+		      }
 		  }
 		;
